signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/view/SignInWithAppleButton.kt

```diff
@@ -1,28 +1,54 @@
 package com.willowtreeapps.signinwithapplebutton.view
 
-import android.R.style.Theme_Black_NoTitleBar_Fullscreen
-import android.annotation.SuppressLint
-import android.app.Dialog
 import android.content.Context
 import android.graphics.drawable.GradientDrawable
+import android.os.Parcel
+import android.os.Parcelable
 import android.util.AttributeSet
 import android.util.Log
 import android.view.LayoutInflater
-import android.webkit.WebView
 import android.widget.FrameLayout
 import android.widget.TextView
 import androidx.core.content.ContextCompat
 import com.willowtreeapps.signinwithapplebutton.R
+import com.willowtreeapps.signinwithapplebutton.SignInWithAppleClient
 import com.willowtreeapps.signinwithapplebutton.SignInWithAppleService
+import java.util.*
 
-@SuppressLint("SetJavaScriptEnabled")
 class SignInWithAppleButton @JvmOverloads constructor(
     context: Context, attrs: AttributeSet? = null,
     defStyleAttr: Int = 0, defStyleRes: Int = 0
 ) : FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
 
-    var service: SignInWithAppleService? = null
-    private var dialog: Dialog? = null
+    private class SavedState : BaseSavedState {
+
+        var fragmentTag: String? = null
+
+        constructor(source: Parcel) : super(source) {
+            fragmentTag = source.readString()
+        }
+
+        constructor(superState: Parcelable) : super(superState)
+
+        override fun writeToParcel(out: Parcel?, flags: Int) {
+            super.writeToParcel(out, flags)
+            out?.writeString(fragmentTag)
+        }
+
+        companion object CREATOR : Parcelable.Creator<SavedState> {
+            override fun createFromParcel(parcel: Parcel) = SavedState(parcel)
+            override fun newArray(size: Int): Array<SavedState?> = arrayOfNulls(size)
+        }
+
+    }
+
+    private var service: SignInWithAppleService? = null
+    private var client: SignInWithAppleClient? = null
+    private var fragmentTag: String? = null
+        set(value) {
+            field = value
+            configureFragmentIfCreated()
+        }
```

Comment:
In Swift we have `didSet { … }` for cases like this … is there anything like that on properties or is this the right way to react to changes?